Interact Rework

Maurdeer opened the pull request:
- Created a abstract class IInteractable to replace the differences in logic between each individual collectible and to avoid unnecessary repetition.
- Added trigger colliders to some prefabs to accommodate this change

Maurdeer committed:
Interact Rework

- Created a abstract class IInteractable to replace the differences in logic between each individual collectible and to avoid unnecessary repetition.
- Added trigger colliders to some prefabs to accommodate this change

Carlijavier5 reviewed (commented):
All issues well fixed. Epic job!

Maurdeer committed:
Fixed edge cases in Interact changes

- Fixed edge case where chest spam was possible by reimplementing a former solution
- Fixed edge case where by quickly leaving and reentering the range of a chest the pop up would not appear
- Fixed case where the player could interact during a paused menu

Maurdeer committed:
Fix Player Positioning

I'm so sorry T-T

JungwooJang119 reviewed (approved):
wahoo
